File: N으로표현.java
import java.util.*;
class Solution {
    public int solution(int N, int number) {
        int answer = 0;
        HashSet<Integer> s = new HashSet<>();
        ArrayList<HashSet<Integer>> arr = new ArrayList<>();
        for(int i=0;i<=9;i++){
            arr.add(new HashSet<Integer>());
        }
        for(int i=1;i<9;i++){
            String temp = "";
            for(int j=0;j<i;j++){
                temp += Integer.toString(N);
            }
            arr.get(i).add(Integer.parseInt(temp));
        }
        for(int i=1;i<9;i++){
            for(int j=1;j<i;j++){
                for(int x:arr.get(j)){
                    for(int y:arr.get(i-j)){
                        arr.get(i).add(x+y);
                        arr.get(i).add(x-y);
                        arr.get(i).add(x*y);
                        if(y!=0) arr.get(i).add(x/y);
                    }
                }
            }
            if(arr.get(i).contains(number)){
                answer = i;
                break;
            }
        }
        if (answer==0) answer = -1;
        return answer;
    }
}

File: N으로표현.py
def solution(N, number):
    answer = 0
    s=list(set() for _ in range(9))
    for i in range(1, 9):
        s[i].add(int(str(N)*i))
    for i in range(1, 9):
        for j in range(1, i):
            for x in s[j]:
                for y in s[i-j]: 
                    s[i].add(x+y)
                    s[i].add(x-y)
                    s[i].add(x*y)
                    if y!=0: s[i].add(x//y)
        if number in s[i]: 
            answer = i
            break
    if answer == 0:
        answer = -1
    return answer
